packages/editor/package.json

```diff
@@ -4,8 +4,8 @@
   "description": "Theia - Editor Extension",
   "dependencies": {
     "@theia/core": "^0.2.0",
-    "@theia/preferences": "^0.2.0",
-    "@theia/workspace": "^0.2.0"
+    "@theia/preferences-api": "^0.2.0",
+    "@theia/filesystem": "^0.2.0"
```

Comment:
I think `FileIconProvider` should be moved to `@theia/core` from `@theia/filesystem` and renamed `IconProvider`, otherwise extensions are forced to depend on the filesystem

Reply:
ok

Reply:
An issue for it: https://github.com/theia-ide/theia/issues/801